AirCasting/ABConnector/SDSyncController.swift

```diff
@@ -57,31 +61,65 @@ class SDSyncController: ObservableObject {
                 case .success(let metadata):
                     progress(.finalizing)
                     let files = self.fileWriter.finishAndSave()
-                    self.checkFilesForCorruption(files, expectedMeasurementsCount: metadata.expectedMeasurementsCount)
-                    //TODO: Continue processing SD card data when files are not corrupted. Return an error and finish sync without clearing sd card if they are.
-                    if let mobileFileURL = files.first(where: { $0.1 == SDCardSessionType.mobile })?.0 {
-                        self.mobileSessionsSaver.saveDataToDb(fileURL: mobileFileURL, deviceID: sensorName) { result in
-                            switch result {
-                            case .success(let sessions):
-                                self.averagingService.averageMeasurements(for: sessions) {
-                                    Log.info("Averaging done")
-                                    self.onCurrentSyncEnd { self.startBackendSync() }
-                                }
-                                completion(true)
-                            case .failure(let error):
-                                Log.error("Failed to save sessions to database: \(error.localizedDescription)")
-                                completion(false)
-                            }
+                    self.checkFilesForCorruption(files, expectedMeasurementsCount: metadata.expectedMeasurementsCount) { result in
+                        switch result {
+                        case .success:
+                            self.handle(files: files, deviceID: sensorName, completion: completion)
+                        case .failure(let error):
+                            Log.error("[SD Sync] File corrupted: \(error)")
+                            completion(false)
                         }
-                    } else {
-                        completion(true)
                     }
-                case .failure: self.fileWriter.finishAndRemoveFiles(); completion(false)
+                case .failure:
+                    self.fileWriter.finishAndRemoveFiles()
+                    completion(false)
                 }
             }
         })
     }
     
+    private func handle(files: [(URL, SDCardSessionType)], deviceID: String, completion: @escaping (Bool) -> Void ) {
+        let mobileFileURL = files.first(where: { $0.1 == SDCardSessionType.mobile })?.0
+        let fixedFileURL = files.first(where: { $0.1 == SDCardSessionType.fixed })?.0
+        
+        ///TODO: There's a possibility that we have two files - one mobile and one fixed
+        ///then completion can be called more than once - after processing mobile is done and after processing fixed
+        /// Also we should handle situation when on processing is successfull and one is not.
+        #warning("TODO is important here!")
```

Comment:
Completion should be called only after both steps are done. We can't call completion when saving mobile measurements to db is successful, but uploading fixed measurements is not (for example because the internet is not working). If we did, then we would clear sd card even though we didn't send fixed measurements to db, and those measurements would be lost forever.

Reply:
Done.